knowledge-graph/src/main/scala/io/renku/knowledgegraph/users/projects/finder/GLProjectFinder.scala

```diff
@@ -60,14 +63,20 @@ private class GLProjectFinderImpl[F[_]: Async: GitLabClient: Logger](creatorFind
     case (proj, None) => proj.copy(maybeCreator = None).pure[F]
   }.sequence
 
-  private def findProjectsAndCreators(criteria: Criteria): F[List[ProjectAndCreator]] =
-    GitLabClient[F].get(uri"users" / criteria.userId.value.show / "projects", "user-projects")(
-      mapResponse
-    )(criteria.maybeUser.map(_.accessToken))
+  private def findProjectsAndCreators(criteria: Criteria, page: Page = Page.first): F[List[ProjectAndCreator]] =
+    GitLabClient[F]
+      .get(uri"users" / criteria.userId / "projects" withQueryParam ("page", page), "user-projects")(mapResponse)(
+        criteria.maybeUser.map(_.accessToken)
+      )
+      .flatMap {
+        case (results, None)           => results.pure[F]
+        case (results, Some(nextPage)) => findProjectsAndCreators(criteria, nextPage).map(results ::: _)
```

Comment:
We could also use a `fs2.Stream` to load it lazily (if we don't always need everything immediately). Might also help if GL should have some rate-limiting in place? Or should we handle this (better) explicitely?

Reply:
Yes, we could use streams here but actually, we need to fetch all the user projects as we then combine them with kg projects and only after we sort and page them on the Scala side. So I thought maybe it was not worth switching to Stream here.